### DISCUSSION TITLE
Modelica-GNU_ScientificLibrary update

### DISCUSSION THREAD

**Discussion by tburch-LSS** [OpenModelica/OpenModelica#11299] [General]
Hello there... just wanted to point out to the OM developers that the extension package for GSL has been updated (GSL v2.7 and tested with within OpenModelica 1.21.0) and that the main branch for this library has moved to:

https://github.com/ORNL-Modelica/Modelica-GNU_ScientificLibrary

thanks and carry on!

**Comment by casella** [comment]
@mahge do you want to take care of that?

**Reply by mahge** [reply]
Take care of what? 🙂. If it is not there already we should perhaps add it to the list of libraries handled by the package manager and the regression testing. 

@tburch-LSS would you like us to do that?

**Reply by tburch-LSS** [reply]
it is in the package manager already. the only change that is required from the OM side would be to update the master branch location to the one above. in the meantime, i figured out the proper json file to affect this change, but it would probably be best if you did it... thanks!

**Comment by casella** [comment]
@mahge I'm sorry I misread that. When I saw GSL v2.7 I automatically thought about runtime libraries 😅 

Of course we should add this to the package manager and regression testing. I'll take care of that ASAP.

**Comment by casella** [comment]
Package manager location updated: https://github.com/OpenModelica/OMPackageManager/commit/bc93750692a0daa0372f1b191d345d9b0c220d45

@tburch-LSS, you wrote it's tested with OMC 1.21.0. Do you mean that you tested all models manually and they all work? If that is the case, we could turn the level of support from "experimental" to "full support" and then also test it in our [library testsuite](https://libraries.openmodelica.org/branches/overview.html), so the support status is publicly visible and we get regression reports if we break something in OMC that causes some models in it to fail.

I would also suggest that you add a version annotation, tag the library and release it using GitHub's release management features. This allows better package management, version-less libraries are a bit fishy for that.

**Reply by tburch-LSS** [reply]
super!
yes, manual testing within OMEdit and all appears to function properly. the only outstanding issue might be the windows-compiled GSL libraries in the Resources folder (the linux ones are in order). i will have to confer with the other contributors about this and about versioning (i am not the owner of the master branch).
i would also be up for the support upgrade. thanks for the help and advice!

**Reply by casella** [reply]
I guess before claiming full support we need to have the libraries in place also for Windows. Let me know when that is done. Until then, I'd keep it experimental.

**Reply by casella** [reply]
For CI testing, it would be best to have a version number, then I can set it up in no time.
